reactive/kotlinx-coroutines-rx2/test/SchedulerTest.kt

```diff
@@ -31,4 +38,252 @@ class SchedulerTest : TestBase() {
         }
         finish(4)
     }
+
+    @Test
+    fun testAsSchedulerWithNoDelay(): Unit = runTest {
+        expect(1)
+        val mainThread = Thread.currentThread()
+        val scheduler = (currentDispatcher() as CoroutineDispatcher).asScheduler()
+        suspendCancellableCoroutine<Unit> {
+            scheduler.scheduleDirect {
+                val t1 = Thread.currentThread()
+                assertSame(t1, mainThread)
```

Comment:
It's not clear these tests actually verify anything about threading. It would be better to use an explicit dispatcher that is guaranteed to use a different thread from the one the unit tests are being ran. Otherwise these tests will pass even if you remove the launch calls.